upstairs/src/lib.rs

```diff
@@ -86,17 +83,20 @@ mod downstairs;
 mod upstairs;
 use upstairs::{UpCounters, UpstairsAction};
 
+mod io_limits;
+use io_limits::IOLimitGuard;
+
 /// Max number of write bytes between the upstairs and an offline downstairs
 ///
-/// If we exceed this value, the upstairs will give
-/// up and mark the offline downstairs as faulted.
-const IO_OUTSTANDING_MAX_BYTES: u64 = 1024 * 1024 * 1024; // 1 GiB
+/// If we exceed this value, the upstairs will give up and mark the offline
+/// downstairs as faulted.
+const IO_OUTSTANDING_MAX_BYTES: u64 = 50 * 1024 * 1024; // 50 MiB
 
 /// Max number of outstanding IOs between the upstairs and an offline downstairs
 ///
 /// If we exceed this value, the upstairs will give up and mark that offline
 /// downstairs as faulted.
-pub const IO_OUTSTANDING_MAX_JOBS: usize = 10000;
+pub const IO_OUTSTANDING_MAX_JOBS: usize = 1000;
```

Comment:
I'm curious to see how 1000 jobs does with replay tests.
We might need to change a few things around in the tests to be sure they stick under this limit, otherwise we are testing live repair.

Reply:
Good catch, this was left in by accident.

I didn't see any performance differences, though, so I'm tempted to stick with the smaller numbers...

Reply:
I'm thinking more about when we can still replay vs. LiveRepair.  If this gives us a little more time before we give up on a downstairs, that should result in less work when it comes back.

Reply:
I think we're in the clear: in the [most recent buildomat logs for `test_replay`](https://buildomat.eng.oxide.computer/wg/0/artefact/01JBCZMVYBQTGG2HBRSYCVXT0Q/FG5ETDGnFlNC9JUCKj8scbyHSSbqVZ6JUvlECF0ZFdbOLvaU/01JBCZNGW1ZQ9EP560WE4FBSXJ/01JBD13WSNDTA8MN8SB302NZ0X/test_replay.log), there are indications that it's going through replay (and not live-repair)

```json
{"msg":"[2] client re-new 302 jobs since flush 3114","v":0,"name":"crucible","level":30,"time":"2024-10-29T20:50:36.963940864Z","hostname":"w-01JBD0H1N4K3YEM72S3XWCQD40","pid":1198,"":"downstairs","":"downstairs","session_id":"18a6fd2d-1f19-4552-b9d6-5ab6b53fbb51"}
{"msg":"[2] Replayed 302 jobs since flush: 3114","v":0,"name":"crucible","level":30,"time":"2024-10-29T20:50:36.964171465Z","hostname":"w-01JBD0H1N4K3YEM72S3XWCQD40","pid":1198,"":"downstairs","":"downstairs","session_id":"18a6fd2d-1f19-4552-b9d6-5ab6b53fbb51"}
```